src/Perpetuum/Services/EventServices/EventProcessors/NpcStateAnnouncer.cs

```diff
@@ -51,10 +52,25 @@ private int GetNpcDef(NpcStateMessage msg)
         private string GetNpcName(int def)
         {
             var nameToken = EntityDefault.Get(def).Name + "_name";
-            var name = _nameDictionary[nameToken]?.ToString();
+            var name = "";
+            if (_nameDictionary.TryGetValue(nameToken, out name) != true)
+            {
+                WriteNPCStateAnnouncerLog($"Missing localized name for {nameToken}");
+            }
             return name ?? string.Empty;
```

Comment:
You can actually flip this all around.  But one nit:
`_nameDictionary.TryGetValue(nameToken, out name) != true` can just be `!_nameDictionary.TryGetValue(nameToken, out name)`.

As for the flip:
```cs
if (_nameDictionary.TryGetValue(nameToken, out var name))
{
    return name;
}
// "else" or fall-through when tryget returns false (and var name is not needed in this scope)
WriteNPCStateAnnouncerLog($"Missing localized name for {nameToken}");
return string.Empty;
```

Reply:
Or something like that, I didn't mock this up in the IDE.  But TryGet and then do things in the scope of the if(true) case is where it really shines semantically

Reply:
Since the existing return statement was already offering 'name' or an empty string, I figured I would keep that and work around it. My thought is that fewer return statements make it easier to read and we might end up with multiple checks as other bugs are discovered.